Modules/StudyProgramme/classes/class.ilObjStudyProgrammeMembersGUI.php

```diff
@@ -726,61 +676,74 @@ protected function removeUserMulti() : void
         if (count($not_removed) == count($prgrs_ids)) {
             $this->showInfoMessage("remove_users_not_possible");
         } elseif (count($not_removed) > 0) {
-            $this->showSuccessMessage("remove_users_partitial_success");
+            $this->showSuccessMessage("remove_users_partial_success");
         } else {
             $this->showSuccessMessage("remove_users_success");
         }
         $this->ctrl->redirect($this, "view");
     }
 
+    protected function confirmRemoveUsers(array $progress_ids) : string
+    {
+        $this->confirmation_gui->setFormAction($this->ctrl->getFormAction($this));
+        $this->confirmation_gui->setHeaderText($this->lng->txt('confirm_to_remove_selected_assignments'));
+        $this->confirmation_gui->setConfirm($this->lng->txt('prg_remove_user'), 'confirmedRemoveUsers');
+        $this->confirmation_gui->setCancel($this->lng->txt('cancel'), 'view');
+
+        foreach ($progress_ids as $progress_id) {
+            $progress = $this->getProgressObject($progress_id);
+            $user = ilObjUser::_lookupFullname($progress->getUserId());
+            $name = $user . ' (' . $progress->getId() . ')';
+            
+            $this->confirmation_gui->addItem(
+                self::F_SELECTED_PROGRESS_IDS . '[]',
+                $progress_id,
+                $name
+            );
+        }
+        return $this->confirmation_gui->getHTML();
+    }
+
     /**
-     * Rmeove user
+     * Remove user
      */
     protected function remove(int $prgrs_id) : void
     {
         $prgrs = $this->getProgressObject($prgrs_id);
-        $usr_id = $prgrs->getUserId();
-        if (
-            $this->object->getAccessControlByOrguPositionsGlobal() &&
-            !in_array($usr_id, $this->manageMembers()) &&
-            !$this->mayManageMembers()
-        ) {
+
+        if (!in_array(
+            $prgrs->getUserId(),
+            $this->permissions->getUserIdsSuscepibleTo(ilOrgUnitOperation::OP_MANAGE_MEMBERS)
+        )) {
+            var_dump($this->permissions->getUserIdsSuscepibleTo(ilOrgUnitOperation::OP_MANAGE_MEMBERS));
+            die('why not');
```

Comment:
Remove these debugging statements, maybe this should be a `\LogicException` instead?